### PR TITLE
Saw the .NET Foundation project spotlight - felt inclined to contribute

### PR DESCRIPTION
**Pull request by IEvangelist**
## Description

I saw that this project was recently added to the .NET Foundation, congrats! I watched the project spotlight recently, and heard a call to action where @troyhunt mentioned that he'd love more contributors helping.

## In this pull request

This pull request was based on C# observations that I made while reading the code. The improvements and updates are as follows:

- Remove `BlobStorageOption` as a class-scoped field from `BlogStorage` implementation as it's only used in the `.ctor`.
- Remove `sw.Start()` call, as it is redundant - the stop watch starts automatically when calling `Stopwatch.StartNew()`.
- The `BlobStorageEntry` is defined as a positional `record` but has redundant properties, simplified and ensured that the IL is the same.
- In `Hash.IsHexStringOfLength`  add `null` check as it was not guarded against - added corresponding test case to `Theory`.
- Added `HttpRequestDataExtensions` to simplify service resolution in function methods.
- Use top-level program for Azure function entry point.
  - Apply `using` to `host` instance.
  - Use `await host.RunAsync()` instead of `host.Run()`.
- In `Range` remove unused `using` which is a result of the new extension methods.
  - Adhere to the "law of demeter (LoD)", instead of stepping into multiple nested objects with `.` notation - encapsulate it in extension method.

## Checklist:

- [x] I confirm that my submission adheres to the [Code of Conduct](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/CODE_OF_CONDUCT.md).
- [x] I have tested that the code works with sample data and verified all tests are passing
- [x] I have verified that any tests that I have supplied with this Pull Request work and I have fixed any code which caused pre-existing tests to fail.
- [x] I have updated the [README](https://github.com/HaveIBeenPwned/PwnedPasswordsAzureFunction/blob/main/README.md) with any changes in dependencies/configuration values.
- [x] I have linted my code to ensure that it is formatted correctly.

### CODE REVIEW

File: Functions/Hash.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static string CreateSHA1Hash(this string input, string source = "UTF8")
 
         public static bool IsHexStringOfLength(this string input, int requiredLength)
         {
-            if (input.Length == 0 || input.Length != requiredLength)
+            if (input is null || input.Length == 0 || input.Length != requiredLength)
```
Comment (stebet) [inline Functions/Hash.cs:36]: Null check is redundant since it can't currently be called with a null value.

File: Functions.Tests/HashTests.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@ namespace Functions.Tests
     public class HashTests
     {
         [Theory]
+        [InlineData(null, false)]
```
Comment (stebet) [inline Functions.Tests/HashTests.cs:8]: Redundant test since the method will never be called with a null value.

**Review by stebet — COMMENTED** [review]
Like the simplification via. extension methods :)

File: Functions.Tests/HashTests.cs (hunk 2 above)
Comment (stebet) [reply Functions.Tests/HashTests.cs:8]:
```suggestion
```

File: Functions/Hash.cs (hunk 1 above)
Comment (stebet) [reply Functions/Hash.cs:36]:
```suggestion
            if (input.Length == 0 || input.Length != requiredLength)
```

**Review by stebet — CHANGES_REQUESTED** [review]
I made a couple of suggestion @IEvangelist :)

Reason we can skip the null checks as that this method will never by called with a null value since the `Range` function won't be called with a null range parameter (or it wouldn't bind to it anyway). We also have `nullable` checks enabled for the solution to grab other cases.

**Comment by IEvangelist** [comment]
> I made a couple of suggestion @IEvangelist :)
> 
> Reason we can skip the null checks as that this method will never by called with a null value since the `Range` function won't be called with a null range parameter (or it wouldn't bind to it anyway). We also have `nullable` checks enabled for the solution to grab other cases.

Hi @stebet - awesome, I've accepted the suggestions and updated the PR. Thank you.

**Pull request merged**